File: solutions/go/vehicle-purchase/1/vehicle_purchase.go
package purchase

import "fmt"

func NeedsLicense(kind string) bool {
	return kind == "car" || kind == "truck"
}

func ChooseVehicle(option1, option2 string) string {
    var bestChoice string
    if option1 < option2{
		bestChoice = option1
    } else {
    	bestChoice = option2
    }
	return fmt.Sprintf("%s is clearly the better choice.", bestChoice)
}

func CalculateResellPrice(originalPrice, age float64) float64 {
	if age < 3 {
        return originalPrice * 0.8
    } else if age < 10 {
        return originalPrice * 0.7
    } else {
		return originalPrice * 0.5
    }
}

File: solutions/javascript/vehicle-purchase/1/vehicle-purchase.js
export function needsLicense(kind) {
  if (kind === "car" || kind === "truck"){
    return true    
  }else{
    return false
  }
}

export function chooseVehicle(option1, option2) {
  if (option1[0] < option2[0]){
     return option1 + ' is clearly the better choice.'
  }
  return option2 + ' is clearly the better choice.'
}

export function calculateResellPrice(originalPrice, age) {
  let disccount = 0
  if (age < 3){disccount = 0.8}
  else if (age > 10) {disccount = 0.5}
  else {disccount = 0.7}
  return originalPrice * disccount;
}
